dynamiqs/qarrays/sparse_dia_qarray.py

```diff
@@ -250,8 +250,14 @@ def _matmul_dia(self, other: SparseDIAQArray) -> QArray:
     def __and__(self, y: QArray) -> QArray:
         return NotImplemented
 
-    def __pow__(self, y: Scalar) -> QArray:
-        return NotImplemented
+    def powm(self, n: ScalarLike) -> QArray:
+        result = self
+        for _ in range(n - 1):
+            result = result @ self
```

Comment:
```suggestion
        f = lambda carry, _: (carry @ self, None)
        jax.lax.scan(f, self, length=n - 1)
```
(to shorten the jit time)